R/dependencies.R

```diff
@@ -631,3 +618,62 @@ fileDependencies.Rmd.evaluate <- function(file) {
 
   unique(unlist(deps, recursive = TRUE))
 }
+
+
+
+
+
+fileDependencies.Rmd.tangle <- function(file, encoding = "UTF-8") {
+
+  # discovered packages
+  deps <- list()
+
+  # unique key (line) to split R code with
+  key <- paste0("###--packrat", as.integer(Sys.time()), "\n")
+
+  # rudely override knitr's 'label_code' function so
+  # that we can detect dependencies within inline chunks
+  knitr <- asNamespace("knitr")
```

Comment:
In the future if this pattern is used again, it would be nice to have something like:


```R
label_code <- yoink("knitr", "label_code")
label_code_shim <- function(...) {
  # paste a known key to the end to split the code chunks with
  paste0(key, label_code(...))
}
with_shim(knitr, "label_code", label_code_shim, {
   # Code here
})